include/bitcoin/bitcoin/utility/resource_lock.hpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2011-2015 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * libbitcoin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License with
+ * additional permissions to the one published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version. For more information see LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_FILE_LOCK_HPP
+#define LIBBITCOIN_FILE_LOCK_HPP
+
+#include <boost/filesystem.hpp>
+#include <boost/interprocess/sync/file_lock.hpp>
+
+namespace libbitcoin {
+
+typedef boost::filesystem::path file_path;
+
+/**
+ * A resource lock device that ensures exclusive access to a resource.
+ * It takes a path for creating a lock file object that can be used
+ * for telling the usage of some resource between processes.
+ * Example: opening/closing blockchain database.
+ */
+class resource_lock
```

Comment:
It was called file_lock before, but I tried to more accurately express what the purpose of this class. I have another class for process_lock so that name is a no go. resource expresses that some resources are being locked but file_lock is also possible.